apps/root/src/pages/earn/components/wizard/components/wizard-selection/index.tsx

```diff
@@ -75,11 +77,36 @@ export const WizardSelection = ({
   const intl = useIntl();
   const { mergedBalances, isLoadingAllBalances } = useMergedTokensBalances(ALL_WALLETS);
   const { hasFetchedAllStrategies } = useAllStrategies();
+  const { assetTokenId, rewardTokenId } = useParams<{
+    assetTokenId?: string;
+    rewardTokenId?: string;
+  }>();
+  const replaceHistory = useReplaceHistory();
 
   const isLoading = isLoadingAllBalances || !hasFetchedAllStrategies;
 
   const { assets, rewards } = useStrategiesParameters();
 
+  React.useEffect(() => {
+    const foundAsset = assetTokenId ? findEquivalentTokenById(assets, assetTokenId.toLowerCase()) : undefined;
+    const foundReward = rewardTokenId ? findEquivalentTokenById(rewards, rewardTokenId.toLowerCase()) : undefined;
+
+    if (foundAsset) {
+      setSelectedAsset({
+        key: foundAsset.address,
```

Comment:
carefull this will trigger each time that assetss and rewards are modified, which means, each time we'll be modifying a strategy. Maybe we should check here that if they are not set THEN we set them